File: src/main/java/rbq/wtf/lycoris/client/wrapper/wrappers/wrapper/render/texture/DynamicTexture.java
package rbq.wtf.lycoris.client.wrapper.wrappers.wrapper.render.texture;

import rbq.wtf.lycoris.client.wrapper.MapEnum;
import rbq.wtf.lycoris.client.wrapper.wrappers.annotation.WrapClass;
import rbq.wtf.lycoris.client.wrapper.wrappers.annotation.WrapperClass;
import rbq.wtf.lycoris.client.wrapper.wrappers.utils.ReflectUtil;

import java.awt.image.BufferedImage;

@WrapperClass(mcpName = "net.minecraft.client.renderer.texture.DynamicTexture", targetMap = MapEnum.VANILLA189)
public class DynamicTexture extends AbstractTexture {
    @WrapClass(mcpName = "net.minecraft.client.renderer.texture.DynamicTexture", targetMap = MapEnum.VANILLA189)
    public static Class DynamicTextureClass;

    public DynamicTexture(Object obj) {
        super(obj);
    }

    public DynamicTexture(BufferedImage bufferedImage) {
        super(ReflectUtil.construction(DynamicTextureClass, bufferedImage));
    }

    public DynamicTexture(int textureWidth, int textureHeight) {
        super(ReflectUtil.construction(DynamicTextureClass, textureWidth, textureHeight));
    }
}

File: src/main/java/rbq/wtf/lycoris/client/wrapper/wrappers/wrapper/render/texture/DynamicTexture.kt
package rbq.wtf.lycoris.client.wrapper.wrappers.wrapper.render.texture

import rbq.wtf.lycoris.client.wrapper.MapEnum
import rbq.wtf.lycoris.client.wrapper.wrappers.annotation.WrapClass
import rbq.wtf.lycoris.client.wrapper.wrappers.annotation.WrapperClass
import rbq.wtf.lycoris.client.wrapper.wrappers.utils.ReflectUtil
import java.awt.image.BufferedImage

@WrapperClass(mcpName = "net.minecraft.client.renderer.texture.DynamicTexture", targetMap = MapEnum.VANILLA189)
class DynamicTexture(obj: Any) : AbstractTexture(obj) {
    constructor(bufferedImage: BufferedImage) : this(ReflectUtil.construction(DynamicTextureClass, bufferedImage))
    constructor(textureWidth: Int, textureHeight: Int) : this(
        ReflectUtil.construction(
            DynamicTextureClass,
            textureWidth,
            textureHeight
        )
    )

    companion object {
        @WrapClass(mcpName = "net.minecraft.client.renderer.texture.DynamicTexture", targetMap = MapEnum.VANILLA189)
        lateinit var DynamicTextureClass: Class<*>
    }
}